hack/configapi/cmd/root.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/internal/api/configapi"
+	"github.com/edgelesssys/constellation/v2/internal/staticupload"
+	"github.com/spf13/cobra"
+)
+
+const (
+	awsRegion      = "eu-central-1"
+	awsBucket      = "cdn-constellation-backend"
+	invalidDefault = 0
+	envAwsKeyID    = "AWS_ACCESS_KEY_ID"
+	envAwsKey      = "AWS_ACCESS_KEY"
+)
+
+var (
+	versionFilePath string
+	// Cosign credentials.
+	cosignPwd      string
+	privateKeyPath string
+)
+
+// Execute executes the root command.
+func Execute() error {
+	return NewRootCmd().Execute()
+}
+
+// NewRootCmd creates the root command.
+func NewRootCmd() *cobra.Command {
+	uploadCmd := &cobra.Command{
+		Use:   "AWS_ACCESS_KEY_ID=$ID AWS_ACCESS_KEY=$KEY upload -b 2 -t 0 -s 6 -m 93 --cosign-pwd $PWD --private-key $FILE_PATH",
+		Short: "Upload a set of versions specific to the azure-sev-snp attestation variant to the config api.",
+
+		Long: "Upload a set of versions specific to the azure-sev-snp attestation variant to the config api. Please authenticate with AWS through your preferred method (e.g. environment variables, CLI) to be able to upload to S3.",
+		RunE: runCmd,
+	}
+	uploadCmd.PersistentFlags().StringVarP(&versionFilePath, "version-file", "f", "", "File path to the version json file.")
+
+	uploadCmd.PersistentFlags().StringVar(&cosignPwd, "cosign-pwd", "", "Cosign password used to decrpyt the private key.")
+
+	uploadCmd.PersistentFlags().StringVar(&privateKeyPath, "private-key", "", "File path of private key used to sign the payload.")
+	return uploadCmd
+}
+
+func runCmd(cmd *cobra.Command, _ []string) error {
+	if err := enforceRequiredFlags(cmd, "version-file", "cosign-pwd", "private-key"); err != nil {
+		return err
+	}
```

Comment:
how to handle the error then? should we panic? i wanted to avoid that. what is the benefit?

Reply:
`MarkFlagRequired` only returns an error if the flag one is asking to mark as required [does not actually exist](https://github.com/spf13/pflag/blob/2e9d26c8c37aae03e3f9d4e90b7116f5accb7cab/flag.go#L495-L498).
Since this should never be the case if we do so directly after we define the flag, we commonly wrap the call in a function that panics on error.
See the CLI [`must` function for reference](https://github.com/edgelesssys/constellation/blob/c5e016a8e28fef42a5cc1fa758676caed773902f/cli/cmd/root.go#L99).

We want to mark the flag as required in the `NewRootCmd` function so the autocomplete directives, and error handling if not set, work before we actually execute the command.
This wont work (or shouldnt work) if we only mark them as required if the command is already executed.